bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ComponentFilesImpl.java

```diff
@@ -27,24 +28,33 @@
 import javax.inject.Named;
 
 import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.LoginException;
 import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
 import org.apache.sling.api.resource.ResourceResolverFactory;
 import org.apache.sling.models.annotations.Default;
 import org.apache.sling.models.annotations.Model;
 import org.apache.sling.models.annotations.injectorspecific.OSGiService;
-import org.apache.sling.models.annotations.injectorspecific.Self;
 
 import com.adobe.cq.wcm.core.components.internal.Utils;
 import com.adobe.cq.wcm.core.components.models.ComponentFiles;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Model(
         adaptables = SlingHttpServletRequest.class,
         adapters = ComponentFiles.class
 )
 public class ComponentFilesImpl implements ComponentFiles {
+    private static final Logger LOG = LoggerFactory.getLogger(ComponentFilesImpl.class);
 
-    @Self
-    SlingHttpServletRequest request;
+    /**
+     * Name of the subservice used to authenticate as in order to be able to read details about components and
+     * client libraries.
+     */
+    public static final String COMPONENTS_SERVICE = "components-service";
```

Comment:
We could reuse previously defined constant.

Reply:
We could.
The duplication was mostly an artifact of the refactoring.
However, I decided to leave it because the `ComponentFilesImpl.COMPONENTS_SERVICE` subservice and the `ClientLibrariesImpl.COMPONENTS_SERVICE` could potentially be different subservices and i didn't want to encourage a single catch-all subservice to be used across the entire project.

The constant probably should have been renamed. The clearest way might be to actually change the subservice names to something like "componentfiles-service" and "clientlibraries-service", and then provide two mappings to indicate these are different services, that just happen to have the same permissions.